### PR TITLE
CodeQuality Collector name can be set using create request

### LINKED ISSUE #2577: Name of the code-quality create api end point is hard-coded to 'Sonar'
We need to integrate a custom enterprise code quality tool into Hygieia through its api.

By looking at the [code here](https://github.com/Hygieia/Hygieia/blob/master/api/src/main/java/com/capitalone/dashboard/service/CodeQualityServiceImpl.java#L169) the collector's name has been hard-coded to __Sonar__.

I think the correct way to take the collector name, _optionally_, thorough api by adding a field to `CodeQualityCreateRequest`

### PR DESCRIPTION
**Pull request by isuru89**
Resolves #2577
with help of pull request to [hygieia-core PR-#36](https://github.com/Hygieia/hygieia-core/pull/36)

* When creating code quality collector the name is set to the specified in the `CodeQualityCreateRequest#toolName`.
* If it is not specified, then the default tool name `Sonar` will be kept (to tally with existing collectors)

### CODE REVIEW

**Comment by rvema** [comment]
@isuru89 please fix the build errors that resulted as part of this PR

**Comment by isuru89** [comment]
@rvema 
As discussed [here](https://github.com/Hygieia/Hygieia/issues/2577#issuecomment-444191714), this build will fail until [PR-#36](https://github.com/Hygieia/hygieia-core/pull/36) is getting merged into the [Hygieia-core](https://github.com/Hygieia/hygieia-core).

First PR-36 in hygieia-core must be merged, and then this must be.

File: api/src/main/java/com/capitalone/dashboard/service/CodeQualityServiceImpl.java [hunk 1]
```diff
@@ -167,9 +167,13 @@ public String createV2(CodeQualityCreateRequest request) throws HygieiaException
 
     }
 
-    private Collector createCollector() {
+    private Collector createCollector(CodeQualityCreateRequest request) {
         CollectorRequest collectorReq = new CollectorRequest();
-        collectorReq.setName("Sonar");  //for now hardcode it.
+        if (request.getToolName() != null && !request.getToolName().trim().isEmpty()) {
```
Comment (rvema) [inline api/src/main/java/com/capitalone/dashboard/service/CodeQualityServiceImpl.java]: Please use StringUtils utility methods

File: api/src/main/java/com/capitalone/dashboard/service/CodeQualityServiceImpl.java [hunk 2]
```diff
@@ -167,9 +167,13 @@ public String createV2(CodeQualityCreateRequest request) throws HygieiaException
...
+            collectorReq.setName(request.getToolName());
```
Comment (rvema) [inline api/src/main/java/com/capitalone/dashboard/service/CodeQualityServiceImpl.java]: Also try replacing the if condition with ternary operator

**Pull request merged**